### PR TITLE
Add hash field to QueryResponse proto object

### PR DESCRIPTION
**Pull request by luckychess**
## What is this pull request?
Pull request adds hash field to QueryResponse proto object.
   
## Why do you implement it? Why do we need this pull request?
According our model QueryResponse always contains a hash of a corresponding query but it was not reflected in proto scheme.

## Details/Features
- Add hash field in QueryResponse proto
- Hash field is set in QueryService
- Tests know about hash field and check it

## P.S. (optional)
Since every QueryResponse hash should be the same as hash of corresponding query I decided not to create additional test for hash but add checks for every existing test similar to status call checking.

### CODE REVIEW

File: irohad/torii/impl/query_service.cpp [hunk 1]
```diff
@@ -40,29 +40,29 @@ namespace torii {
     });
   }
 
-  void QueryService::FindAsync(iroha::protocol::Query const& request,
-                               iroha::protocol::QueryResponse& response) {
+  void QueryService::FindAsync(iroha::protocol::Query const &request,
+                               iroha::protocol::QueryResponse &response) {
     // Get iroha model query
     auto query = pb_query_factory_->deserialize(request);
+    auto hash = iroha::hash(**query).to_string();
```
Comment (kamilsa) [inline irohad/torii/impl/query_service.cpp]: Should not we check here if deserialization returned value of none?

File: schema/responses.proto [hunk 2]
```diff
@@ -70,14 +70,15 @@ message TransactionsResponse {
...
     oneof response {
-        AccountAssetResponse account_assets_response = 1;
-        AccountResponse account_response = 2;
-        ErrorResponse error_response = 3;
-        SignatoriesResponse signatories_response = 4;
-        TransactionsResponse transactions_response = 5;
-        AssetResponse asset_response = 6;
-        RolesResponse roles_response = 7;
-        RolePermissionsResponse role_permissions_response = 8;
+        AccountAssetResponse account_assets_response = 2;
```
Comment (kamilsa) [inline schema/responses.proto]: Should start with 1

File: irohad/torii/impl/query_service.cpp (hunk 1 above)
Comment (l4l) [reply irohad/torii/impl/query_service.cpp]: Definitely, at least `BOOST_STATIC_ASSERT`

Comment (luckychess) [reply irohad/torii/impl/query_service.cpp]: Well, I agree but do you think it should be done in this pr's scope?

File: irohad/torii/impl/query_service.cpp [hunk 3]
```diff
@@ -40,29 +40,34 @@ namespace torii {
...
 
-  void QueryService::FindAsync(iroha::protocol::Query const& request,
-                               iroha::protocol::QueryResponse& response) {
+  void QueryService::FindAsync(iroha::protocol::Query const &request,
+                               iroha::protocol::QueryResponse &response) {
     // Get iroha model query
     auto query = pb_query_factory_->deserialize(request);
+    if (not query) {
+      response.mutable_error_response()->set_reason(
+          iroha::protocol::ErrorResponse::WRONG_FORMAT);
```
Comment (muratovv) [inline irohad/torii/impl/query_service.cpp]: Maybe use monadic bind for this code?

File: schema/responses.proto [hunk 4]
```diff
@@ -80,4 +80,5 @@ message QueryResponse {
         RolesResponse roles_response = 7;
         RolePermissionsResponse role_permissions_response = 8;
     }
+    bytes query_hash = 9;
```
Comment (muratovv) [inline schema/responses.proto]: I guess, more convenient to name as just `hash` because of the context `Query` already.

**Pull request merged**